Makefile

```diff
@@ -331,77 +377,133 @@ test-docker: linux go-build-image
+		go test -v -timeout=7m ./test/metrics \
+		--target-image $(CONSUL_COMPAT_TEST_IMAGE) \
+		--target-version local \
+		--latest-image $(CONSUL_COMPAT_TEST_IMAGE) \
+		--latest-version latest
 
-# Build the static web ui inside a Docker container. For local testing only; do not commit these assets.
-ui: ui-docker
+.PHONY: test-connect-ca-providers
+test-connect-ca-providers: ## Running /agent/connect/ca tests in verbose mode
+	@echo "Running /agent/connect/ca tests in verbose mode"
+	@go test -v ./agent/connect/ca
+	@go test -v ./agent/consul -run Vault
+	@go test -v ./agent -run Vault
+
+##@ UI
+
+.PHONY: ui
+ui: ui-docker ## Build the static web ui inside a Docker container. For local testing only; do not commit these assets.
 
-# Build the static web ui with yarn. This is the version to commit.
 .PHONY: ui-regen
-ui-regen:
+ui-regen: ## Build the static web ui with yarn. This is the version to commit.
 	cd $(CURDIR)/ui && make && cd ..
 	rm -rf $(CURDIR)/agent/uiserver/dist
 	mv $(CURDIR)/ui/packages/consul-ui/dist $(CURDIR)/agent/uiserver/
 
-tools:
+.PHONY: ui-build-image
+ui-build-image: ## Building UI build container
+	@echo "Building UI build container"
+	@docker build $(NOCACHE) $(QUIET) -t $(UI_BUILD_TAG) - < build-support/docker/Build-UI.dockerfile
+
+.PHONY: ui-docker
+ui-docker: ui-build-image ## Builds ui within docker container and copy all the relevant artifacts out of the containers back to the source
+	@$(SHELL) $(CURDIR)/build-support/scripts/build-docker.sh ui
+
+##@ Tools
+
+.PHONY: tools
+tools: ## Installs various supporting Go tools.
 	@$(SHELL) $(CURDIR)/build-support/scripts/devtools.sh
 
 .PHONY: lint-tools
-lint-tools:
+lint-tools: ## Install tools for linting
 	@$(SHELL) $(CURDIR)/build-support/scripts/devtools.sh -lint
 
-.PHONY: proto-tools
-proto-tools:
-	@$(SHELL) $(CURDIR)/build-support/scripts/devtools.sh -protobuf
-
 .PHONY: codegen-tools
-codegen-tools:
+codegen-tools: ## Install tools for codegen
 	@$(SHELL) $(CURDIR)/build-support/scripts/devtools.sh -codegen
 
 .PHONY: deep-copy
-deep-copy: codegen-tools
+deep-copy: codegen-tools ## Deep copy
 	@$(SHELL) $(CURDIR)/agent/structs/deep-copy.sh
 	@$(SHELL) $(CURDIR)/agent/proxycfg/deep-copy.sh
 	@$(SHELL) $(CURDIR)/agent/consul/state/deep-copy.sh
 
-version:
+print-%  : ; @echo $($*) ## utility to echo a makefile variable (i.e. 'make print-GOPATH')
+
+.PHONY: module-versions
+module-versions: ## Print a list of modules which can be updated. Columns are: module current_version date_of_current_version latest_version
+	@go list -m -u -f '{{if .Update}} {{printf "%-50v %-40s" .Path .Version}} {{with .Time}} {{ .Format "2006-01-02" -}} {{else}} {{printf "%9s" ""}} {{end}}   {{ .Update.Version}} {{end}}' all
+
+.PHONY: docs
+docs: ## Point your web browser to http://localhost:3000/consul to live render docs from ./website/
+	make -C website
+
+##@ Release
+
+.PHONY: dist
+dist: ## Dist builds binaries for all platforms and packages them for distribution
```

Comment:
Does CRT rely on this make target? If not then we should probably remove it.

Reply:
So far, I haven't found any places referencing this target, so will remove it